x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java

```diff
@@ -403,7 +408,12 @@ public void removeLicense(final ActionListener<PostStartBasicResponse> listener)
             "delete license",
             listener
         );
-        clusterService.submitStateUpdateTask(task.getDescription(), task, newExecutor());
+        clusterService.submitStateUpdateTask(
+            task.getDescription(),
+            task,
+            ClusterStateTaskConfig.build(Priority.NORMAL, masterNodeTimeout),
```

Comment:
NB the master-node timeout is populated in `RestDeleteLicenseAction` but was not being passed to the master service, so previously these tasks would never time out. Clients typically implement their own client-side time out and it can be pretty confusing on a backlogged master since the task it submitted would sit in the queue and eventually be processed. This line addresses that missing plumbing, which now means that by default these requests will time out and properly fail after 30s.

Reply:
Changed my mind - I think we should fix these timeouts but I don't think it's right to do that in a `>refactoring` PR. Reverted in 06fec7b, and I opened https://github.com/elastic/elasticsearch/issues/85996 to track the bug.